samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/facade/AssessmentFacadeQueries.java

```diff
@@ -1933,6 +1933,13 @@ private void updateTitleForCopy(AssessmentData assessmentData, String apepndCopy
     	StringBuffer sb = new StringBuffer(assessmentData.getTitle());
     	sb.append(" ");
     	sb.append(apepndCopyTitle);
+    	if(sb.length() >= assessmentData.TITLE_LENGTH){ //title max size
+    		String appendCopyText = "... "+apepndCopyTitle;
+    		String titleCut = sb.substring(0, assessmentData.TITLE_LENGTH-appendCopyText.length()-1); //cut until size needed to add ellipsis and copyTitle without exceed DB field size
+    		log.debug("titleCut = "+titleCut);
+    		sb = new StringBuffer(titleCut);
+    		sb.append(appendCopyText);
+    	}
```

Comment:
Has this been tested? It doesn't seem correct to me.

titleCut can be 254 characters long.  Plus how long can appendCopyText be? At least two characters, no?